astropy/io/misc/yaml.py

```diff
@@ -223,6 +243,23 @@ def ignore_aliases(self, data):
 AstropyDumper.add_representer(TimeDelta, _timedelta_representer)
 AstropyDumper.add_representer(coords.SkyCoord, _skycoord_representer)
 
+# Numpy dtypes
+AstropyDumper.add_representer(np.bool_,
+                              yaml.representer.SafeRepresenter.represent_bool)
+for np_type in [np.int_, np.int, np.intc, np.intp, np.int8, np.int16, np.int32,
+                np.int64, np.uint8, np.uint16, np.uint32, np.uint64]:
+    AstropyDumper.add_representer(np_type,
+                                 yaml.representer.SafeRepresenter.represent_int)
+for np_type in [np.float_, np.float, np.float16, np.float32, np.float64,
+                np.longdouble]:
```

Comment:
Do we want to include `longdouble` and the `complex128` here?  Couple of concerns:

- Are these types available on all supported platforms?  I had the impression that some 32-bit platforms might simply not have `np.longdouble`, but I'm not sure.
- I'm guessing there will be loss of precision since it round trips back as a `float`.

In all the other cases you don't actually lose information in serializing to YAML, though you do lose the original data type (which is acceptable).